carmm/run/workflows/react.py

```diff
@@ -577,18 +578,19 @@ def _calc_generator(params,
     """New method that gives a default calculator"""
 
     from carmm.run.aims_calculator import get_aims_and_sockets_calculator
+    from ase.calculators.calculator import Parameters
     """On machines where ASE and FHI-aims are run separately (e.g. ASE on login node, FHI-aims on compute nodes)
     we need to specifically state what the name of the login node is so the two packages can communicate"""
     sockets_calc, fhi_calc = get_aims_and_sockets_calculator(dimensions=dimensions,
                                                              logfile=f"{directory}/socketio.log",
                                                              verbose=True,
                                                              codata_warning=False,
-                                                             directory=directory
-                                                             )
+                                                             directory=directory,
+                                                             **params)
 
     """Remove previous xc argument to ensure libxc warning override is first"""
     fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'xc'}
-    fhi_calc.parameters['override_warning_libxc'] ='True'
+    fhi_calc.parameters['override_warning_libxc'] = 'True'
```

Comment:
The `xc` keyword is removed, but never reintroduced, which will result in calculation failure. Can we instead sort the parameters alphabetically rather than removing and reintroducing? The `override_warning_libxc` will then be before `xc` anyway, and the `control.in` file will look a bit neater as a result.

Reply:
Hi Igor. Thanks for your comments. The xc keyword is reintroduced through these line below:
line 604-606 in new code:
```
    """FHI-aims settings set up"""
    for k, v in params.items():
        fhi_calc.parameters[k] = v
```
`params` is not changed from the start, so these lines will reintroduce `xc`.

Reply:
I also tried to implement the sorting of parameters. I just pushed that commit.